src/instana/span/registered_span.py

```diff
@@ -34,6 +44,10 @@ def __init__(self, span, source, service_name, **kwargs) -> None:
         if "gcps" in span.name:
             self.n = "gcps"
 
+        # unify the span name for kafka-producer and kafka-consumer
+        if "kafka" in span.name:
```

Comment:
Can we combine the change of L145 with this if block?
```suggestion
        if "kafka" in span.name:
                self.n = "kafka"
                self._collect_kafka_attributes(span)
```

Reply:
We could, but the methods/functions where `self._collect_kafka_attributes(span)` is called aim to set the `span.data`, not the name.

Let's follow the Clean Code principle that functions must only do one thing. In this case, the functions `_populate_{entry, local, exit}_span_data()`  set the `span.data`.  We can create one specific function to set the span name correctly. 

I plan to refactor this file later to make it cleaner, so I'd like to keep it as is for now.